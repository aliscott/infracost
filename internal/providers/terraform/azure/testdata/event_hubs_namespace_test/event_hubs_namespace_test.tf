provider "azurerm" {
  skip_provider_registration = true
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-notificationhub-resources"
  location = "eastus"
}
resource "azurerm_eventhub_namespace" "standard" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "basic" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "standardwithoutusage" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "capacity" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Standard"
  capacity            = 10

  tags = {
    environment = "Production"
  }
}

resource "azurerm_eventhub_namespace" "basicwithoutusage" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"
  capacity            = 1

  tags = {
    environment = "Production"
  }
}


resource "azurerm_eventhub_namespace" "dedicated" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"
  dedicated_cluster_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/clusters/cluster1"

  tags = {
    environment = "Production"
  }
}


resource "azurerm_eventhub_namespace" "dedicated_retention" {
  name                = "acceptanceTestEventHubNamespace"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  sku                 = "Basic"
  dedicated_cluster_id = "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.EventHub/clusters/cluster1"

  tags = {
    environment = "Production"
  }
}