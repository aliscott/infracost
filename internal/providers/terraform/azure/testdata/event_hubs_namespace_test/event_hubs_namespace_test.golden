
 Name                                                   Monthly Qty  Unit                  Monthly Cost 
                                                                                                        
 azurerm_eventhub_namespace.basic                                                                       
 ├─ Throughput                                                    1  units                        $0.02 
 └─ Ingress event (Basic)                                        10  1M events                    $0.28 
                                                                                                        
 azurerm_eventhub_namespace.basicwithoutusage                                                           
 ├─ Throughput                                                    1  units                        $0.02 
 └─ Ingress event (Basic)                          Monthly cost depends on usage: $0.028 per 1M events  
                                                                                                        
 azurerm_eventhub_namespace.capacity                                                                    
 ├─ Throughput                                                   10  units                        $0.30 
 ├─ Ingress event (Standard)                       Monthly cost depends on usage: $0.028 per 1M events  
 └─ Capture                                                     730  hour                        $73.00 
                                                                                                        
 azurerm_eventhub_namespace.dedicated                                                                   
 ├─ Throughput                                                    1  units                        $6.85 
 └─ Extended retention                             Monthly cost depends on usage: $0.10 per GB          
                                                                                                        
 azurerm_eventhub_namespace.dedicated_retention                                                         
 ├─ Throughput                                                    1  units                        $6.85 
 └─ Extended retention                                            1  GB                           $0.10 
                                                                                                        
 azurerm_eventhub_namespace.standard                                                                    
 ├─ Throughput                                                    1  units                        $0.03 
 ├─ Ingress event (Standard)                                    100  1M events                    $2.80 
 └─ Capture                                                     730  hour                        $73.00 
                                                                                                        
 azurerm_eventhub_namespace.standard_auto_inflate                                                       
 ├─ Throughput                                                   10  units                        $0.30 
 ├─ Ingress event (Standard)                       Monthly cost depends on usage: $0.028 per 1M events  
 └─ Capture                                                     730  hour                        $73.00 
                                                                                                        
 azurerm_eventhub_namespace.standardwithoutusage                                                        
 ├─ Throughput                                                    1  units                        $0.03 
 ├─ Ingress event (Standard)                       Monthly cost depends on usage: $0.028 per 1M events  
 └─ Capture                                                     730  hour                        $73.00 
                                                                                                        
 OVERALL TOTAL                                                                                  $309.57 
----------------------------------
To estimate usage-based resources use --usage-file, see https://infracost.io/usage-file